app/controllers/notifications_controller.rb

```diff
@@ -90,7 +90,19 @@ def mark_read
     if params[:id]
       Notification.read(current_user, [params[:id].to_i])
     else
-      Notification.where(user_id: current_user.id).includes(:topic).where(read: false).update_all(read: true)
+      if types = params[:dismiss_types]&.split(",").presence
+        types.map! do |type|
+          Notification.types[type.to_sym] || (
+            raise Discourse::InvalidParameters.new("invalid notification type: #{type}")
+          )
+        end
+      end
```

Comment:
From an API perspective, I think it is more useful if we can return all the invalid types instead of stopping only at the first one. Otherwise, it may take the API user a few iteration before figuring out all the types that are invalid.

Reply:
done!